src/main/java/org/broadinstitute/hellbender/tools/diagnostics/PrintBGZFBlockInformation.java

```diff
@@ -0,0 +1,189 @@
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import picard.cmdline.programgroups.DiagnosticsAndQCProgramGroup;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * A diagnostic tool that prints information about the compressed blocks in a BGZF format file,
+ * such as a .vcf.gz file.
+ *
+ * The output looks like this:
+ *
+ * Block at file offset 0
+ *     - compressed size: 12932
+ *     - uncompressed size: 65280
+ *
+ * Block at file offset 12932
+ *     - compressed size: 9978
+ *     - uncompressed size: 65280
+ * ...
+ * etc.
+ *
+ * The output can be redirected to a file using the -O option.
+ */
+@ExperimentalFeature
+@CommandLineProgramProperties(
+        summary = "Print information about the compressed blocks in a BGZF format file",
+        oneLineSummary = "Print information about the compressed blocks in a BGZF format file",
+        programGroup = DiagnosticsAndQCProgramGroup.class
+)
+public class PrintBGZFBlockInformation extends CommandLineProgram {
+    private final Logger logger = LogManager.getLogger(PrintBGZFBlockInformation.class);
+
+    @Argument(fullName = "bgzf-file", doc = "The BGZF-format file for which to print block information", optional = false)
+    private String bgzfPathString;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, doc = "File to which to write block information (if not specified, prints to standard output", optional = true)
+    private String output;
+
+    private Path bgzfPath;
+
+    private long streamOffset = 0l;
+
+    private PrintStream outStream;
+
+    @Override
+    protected void onStartup() {
+        super.onStartup();
+
+        bgzfPath = IOUtils.getPath(bgzfPathString);
+
+        if ( ! Files.exists(bgzfPath) ) {
+            throw new UserException("File " + bgzfPathString + " does not exist");
+        }
+
+        if ( ! AbstractFeatureReader.hasBlockCompressedExtension(bgzfPathString) ) {
+            throw new UserException("File " + bgzfPathString + " does not end in a recognized BGZF file extension (" +
+                    AbstractFeatureReader.BLOCK_COMPRESSED_EXTENSIONS + ")");
+        }
+
+        if ( output != null ) {
+            try {
+                outStream = new PrintStream(output);
+            } catch (FileNotFoundException e) {
+                throw new UserException.CouldNotCreateOutputFile(output, "Unable to open output file", e);
+            }
+        } else {
+            outStream = System.out;
+        }
+    }
+
+    @Override
+    protected Object doWork() {
+        try ( InputStream bgzfInputStream = Files.newInputStream(bgzfPath) ) {
+            BGZFBlockMetadata blockInfo = null;
+            while ( (blockInfo = processNextBlock(bgzfInputStream, bgzfPathString)) != null ) {
+                outStream.println(String.format("Block at file offset %d", blockInfo.blockOffset));
```

Comment:
Better `outStream.printf()` for removing the `String.format()`.

Reply:
Done